x/evm/types/types.go

```diff
@@ -643,18 +705,27 @@ func createApproveContractCallParams(sourceChain string, event EventContractCall
 		return nil, err
 	}
 
+	uint256Type, err := abi.NewType("uint256", "uint256", nil)
+	if err != nil {
+		return nil, err
+	}
+
 	arguments := abi.Arguments{
 		{Type: stringType},
 		{Type: stringType},
 		{Type: addressType},
 		{Type: bytes32Type},
+		{Type: bytes32Type},
+		{Type: uint256Type},
 	}
 
 	result, err := arguments.Pack(
 		sourceChain,
 		event.Sender.Hex(),
 		common.HexToAddress(event.ContractAddress),
 		common.Hash(event.PayloadHash),
+		common.Hash(sourceTxID),
+		new(big.Int).SetUint64(sourceEventIndex),
```

Comment:
```suggestion
		sdk.NewUint(sourceEventIndex).BigInt(),
```
Perhaps?

Reply:
I don't think it's necessary. `sdk.NewUint` is just doing exactly what I'm doing here, e.g. SetUint64 from big.Int.